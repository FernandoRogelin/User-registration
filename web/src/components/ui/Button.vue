<script setup>
defineProps({
  variant: {
    type: String,
    default: 'primary'
  }
})
</script>

<template>
  <button class="components-ui-button" :class="`components-ui-button--${variant}`">
    <slot />
  </button>
</template>

<style scoped lang="scss">
.components-ui-button{
  border: 0;
  padding: 0;
  width: 100%;
  cursor: pointer;
  border-radius: 5px;
  padding: 0.4rem 1rem;
  color: var(--mb-white);
  background-color: var(--mb-orange);

  &--secondary {
    color: var(--mb-orange);
    background-color: var(--mb-white);
    border: 1px solid var(--mb-orange);
  }
}
</style>
