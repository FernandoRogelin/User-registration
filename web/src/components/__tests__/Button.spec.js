import { describe, expect, test, afterEach } from "vitest"
import { render, screen, cleanup } from "@testing-library/vue"

import Button from "@/components/ui/Button.vue"

const mockSlot = 'Button component'

afterEach(() => cleanup())

describe('<Button /> ui component', () => {
  test('should render button component and match snapshot', () => {
    const { container } = render(Button, { slots: { default: mockSlot } })

    expect(container.firstChild).toMatchSnapshot()
  })

  test('should render secondary button', () => {
    render(Button, {
      slots: { default: mockSlot },
      props: { variant: 'secondary' }
    })

    const button = screen.getByRole('button', { name: /Button component/i })

    expect(button).toHaveClass('components-ui-button--secondary')
  })
})
