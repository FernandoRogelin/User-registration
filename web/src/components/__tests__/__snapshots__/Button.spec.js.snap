// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> ui component > should render button component and match snapshot 1`] = `
<button
  class="components-ui-button components-ui-button--primary"
  data-v-e3216c45=""
>
  
  Button component
  
</button>
`;
